app/models/dialog_field_association_validator.rb

```diff
@@ -14,15 +14,19 @@ def circular_references(associations)
   private
 
   def initial_paths(associations)
-    associations.flat_map { |key, values| values.map { |value| [key, value] } }
+    associations.flat_map { |key, values| values.map { |value| [key, value] if associations.key?(value) } }.compact
   end
 
   def walk_value_path(fieldname_being_triggered, associations, path)
```

Comment:
Can you explain the use case for the method?  Based on the name I would expect it to enumerate over something and `yield` each one, but that doesn't appear to be the case.

Reply:
We have the initial associations: a => b, b => d, d => a
Then the walk value path for each of the initial paths has to check whether the value is also a key, and points to another field that has associations. There's no way to do this without a while. I get that it's gross. I don't know how to do this better.

Reply:
Here's the example that we worked through
```ruby
def check_for_circular_references(hash, k, collection = [])
  raise KeyAlreadyExistsError, "#{k} already exists in #{collection}" if collection.include?(k)
  collection << k
  hash[k].each do |val|
    check_for_circular_references(hash, val, collection.dup)
  end
  []
end

class KeyAlreadyExistsError < RuntimeError;end

hash = {
  "field_a" => ["field_b", "field_c"],
  "field_b" => ["field_d"],
  "field_d" => ["field_c"],
  "field_c" => []
}

hash.each_key do |k|
  result = check_for_circular_references(hash, k)
  puts "#{result}"
rescue KeyAlreadyExistsError => error
  puts "XXX FAILED WITH #{error.message}"
end
```